### PR TITLE
test: ensure compile takes time

### PR DESCRIPTION
**Pull request by lesam**
Add a sleep since part of the test is checking that the compile time is non-zero.

- [ ] [CHANGELOG.md](https://github.com/influxdata/influxdb/blob/master/CHANGELOG.md) updated with a link to the PR (not the Issue)
- [x] [Well-formatted commit messages](https://www.conventionalcommits.org/en/v1.0.0-beta.3/)
- [x] Rebased/mergeable
- [x] Tests pass

### CODE REVIEW

File: query/control/controller_test.go [hunk 1]
```diff
@@ -207,6 +207,8 @@ func TestController_QueryRuntimeError(t *testing.T) {
 
 			q, err := ctrl.Query(context.Background(), &mock.Compiler{
 				CompileFn: func(ctx context.Context) (flux.Program, error) {
+					// ensure we have non-zero compile time
+					time.Sleep(1*time.Millisecond)
```
Comment (lesam) [inline query/control/controller_test.go:211]: I have seen this test fail a few times with the error below: `"expected compile duration to be above zero"`

**Review by codyshepherd — APPROVED** [review]
